File: DSA/Vyom-Singhal/Day 5/Day5-q2.cpp
#include<iostream>
using namespace std;

class Node {
    public :
        int data;
        Node* next;
    void createList(Node* head, int n) {
        Node *temp = new Node();
        int data;
        cout << "Enter data for element 1 : ";
        cin >> data;
        head->data = data;
        head->next = NULL;
        temp = head;
        for(int i = 1; i < n; i++) {
            Node *newNode = new Node();
            cout << "Enter data for element " << i+1 << " : ";
            cin >> data;
            newNode->data = data;
            newNode->next = NULL;
            temp->next = newNode;
            temp = newNode;
        }
    }
    string compareList(Node* head1, Node* head2, int n1, int n2) {
        if(n1 != n2) return "different";
        while(head1 != NULL && head2 != NULL) {
            if(head1->data != head2->data) return "different";
            head1 = head1->next;
            head2 = head2->next;
        }
        return "identical";
    }
    void printList(Node* head) {
        while(head != NULL) {
            cout << head->data << " ";
            head = head->next;
        }
    }
};

int main() {
    Node* head1 = new Node();
    Node* head2 = new Node();
    int n1, n2;
    cout << "Enter number of elements for list 1 : ";
    cin >> n1;
    head1->createList(head1,n1);
    cout << "Enter number of elements for list 2 : ";
    cin >> n2;
    head2->createList(head2,n2);
    cout << head1->compareList(head1,head2,n1,n2);
}

File: DSA/Aanidhay-Aggarwal/Day-5/Day5_q2.java
import java.util.Scanner;

public class LinkedList {
    Node head;

    static class Node {
        int data;
        Node next;

        Node(int data) {
            this.data = data;
            this.next = null;
        }
    }

    public void addNode(int data) {
        Node newNode = new Node(data);
        if (head == null) {
            head = newNode;
        } else {
            Node temp = head;
            while (temp.next != null) {
                temp = temp.next;
            }
            temp.next = newNode;
        }
    }

    public boolean areIdentical(LinkedList list1, LinkedList list2) {
        Node node1 = list1.head;
        Node node2 = list2.head;

        while (node1 != null && node2 != null) {
            if (node1.data != node2.data) {
                return false;
            }
            node1 = node1.next;
            node2 = node2.next;
        }

        if (node1 != null || node2 != null) {
            return false;
        }

        return true;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        LinkedList list1 = new LinkedList();
        System.out.print("Enter the number of elements for list 1: ");
        int n1 = scanner.nextInt();
        System.out.println("Enter the elements for list 1:");
        for (int i = 0; i < n1; i++) {
            list1.addNode(scanner.nextInt());
        }

        LinkedList list2 = new LinkedList();
        System.out.print("Enter the number of elements for list 2: ");
        int n2 = scanner.nextInt();
        System.out.println("Enter the elements for list 2:");
        for (int i = 0; i < n2; i++) {
            list2.addNode(scanner.nextInt());
        }

        LinkedList comparator = new LinkedList();

        System.out.println(comparator.areIdentical(list1, list2) ? "identical" : "different");
    }
}
